index.html

```diff
@@ -126,187 +846,604 @@ <h3>
+                        |promise| with an "{{NetworkError}}" {{DOMException}}.
+                      <li>
+                        If the socket was disconnected, run the following steps:
+                        <ol>
+                          <li>
+                            [=Reject=] |promise| with a "{{NetworkError}}"
+                            {{DOMException}}.
+                          <li>
+                            Invoke the steps to [=handle closing the UDP writable
+                            stream=].
+                        </ol>
+                      <li>
+                        If |signal| is [=AbortSignal/aborted=], [=reject=] |promise|
+                        with |signal|'s [=AbortSignal/abort reason=].
+                    </ol>
+                </ol>
+              <li>
+                Return |promise|.
+            </ol>
+          <li>
+            Let |abortAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Let |promise| be [=a new promise=].
+              <li>
+                Run the following steps [=in parallel=]:
+                <ol>
+                  <li>
+                    Invoke the operating system to close the socket.
+                  <li>
+                    Invoke the steps to [=handle closing the UDP writable stream=].
+                  <li>[=Resolve=] |promise| with `undefined`.
+                </ol>
+              <li>
+                Return |promise|.
+            </ol>
+          <li>
+            Let |closeAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Let |promise| be [=a new promise=].
+              <li>
+                Run the following steps [=in parallel=].
+                <ol>
+                  <li>
+                    Invoke the operating system to close the socket.
+                  <li>
+                    Invoke the steps to [=handle closing the UDP writable stream=].
+                  <li>
+                    If |signal| is [=AbortSignal/aborted=], [=reject=] |promise|
+                    with |signal|'s [=AbortSignal/abort reason=].
+                  <li>Otherwise, [=resolve=] |promise| with `undefined`.
+                </ol>
+              <li>Return |promise|.
+            </ol>
+          <li>
+            [=WritableStream/Set up=] |stream| with
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-writealgorithm">writeAlgorithm</a>
+            set to |writeAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-abortalgorithm">abortAlgorithm</a>
+            set to |abortAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-closealgorithm">closeAlgorithm</a>
+            set to |closeAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-highwatermark">highWaterMark</a>
+            set to 1.
```

Comment:
I think the high water mark should be implementation defined.

Reply:
Fixed. Do we need to have the same logic with TCP writable for coalescing packets?